app/src/main/java/com/example/bigowlapp/utils/NotificationListenerManager.java

```diff
@@ -54,4 +66,36 @@ private void handleAuthByPhoneNumberFailure(AuthByPhoneNumberFailure notificatio
                     .updateDocument(notification.getUid(), notification);
         }
     }
+
+    private void handleSmsInvitationRequest(SmsInvitationRequest notification, Context context){
+        if (notification.isUsed()) {
+            return;
+        }
+        String channel = notification.getTitle();
+        createNotificationChannel(context, channel);
+
+        notificationBuilder(context, channel, notification);
+    }
+
+    private void createNotificationChannel(Context context, String channelId) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            CharSequence name = "Notification";
+            String description = "General notification";
+            int importance = NotificationManager.IMPORTANCE_DEFAULT;
+            NotificationChannel channel = new NotificationChannel(channelId, name, importance);
+            channel.setDescription(description);
+            NotificationManager notificationManager = context.getSystemService(NotificationManager.class);
+            notificationManager.createNotificationChannel(channel);
+        }
+    }
+
+    private void notificationBuilder(Context context, String channel, Notification notification){
+        NotificationCompat.Builder builder = new NotificationCompat.Builder(context, channel)
+                .setSmallIcon(R.drawable.ic_notification)
+                .setContentTitle(channel)
+                .setStyle(new NotificationCompat.InboxStyle()
+                        .addLine(notification.getMessage()));
+        NotificationManagerCompat notificationManager = NotificationManagerCompat.from(context);
+        notificationManager.notify(1, builder.build());
```

Comment:
To make this slightly better, if possible, add an onClick to this notif that goes to the NotificationActivity.